Saule/Serialization/ResourceSerializer.cs

```diff
@@ -235,24 +248,39 @@ private JObject SerializeRelationships(ResourceGraphNode node)
 
             foreach (var kv in node.Relationships)
             {
+                var relationship = kv.Value.Relationship;
+
                 var item = new JObject();
 
                 var data = SerializeRelationshipData(kv.Value);
 
                 var relationshipId = default(string);
 
                 if (data != null
-                    && kv.Value.Relationship.Kind == RelationshipKind.BelongsTo
+                    && relationship.Kind == RelationshipKind.BelongsTo
                     && kv.Value.SourceObject != null)
                 {
                     relationshipId = (string)data["id"];
                 }
 
                 var links = new JObject();
-                AddUrl(links, "self", _urlBuilder.BuildRelationshipPath(node.Resource, node.Key.Id.ToString(), kv.Value.Relationship));
-                AddUrl(links, "related", _urlBuilder.BuildRelationshipPath(node.Resource, node.Key.Id.ToString(), kv.Value.Relationship, relationshipId));
+                var self = _urlBuilder.BuildRelationshipPath(node.Resource, node.Key.Id.ToString(), relationship);
+                var related = _urlBuilder.BuildRelationshipPath(node.Resource, node.Key.Id.ToString(), relationship, relationshipId);
 
-                item["links"] = links;
+                if (!string.IsNullOrEmpty(self) && relationship.LinkType.HasFlag(LinkType.Self))
```

Comment:
I think for resources, we should both check their resources `LinkType` as well as the relationship's `LinkType`. It should probably resolve to the most restricting value.

Exampe:

```
PersonResource: WithLinks(All)
  - BelongsTo<CompanyResource> Job, WithLinks(None)
CompanyResource: WithLinks(Self)
```

Now when serializing the `Job` relationship, what links should show up? If I read this code correctly, currently it will serialize the `self` link even if the `BelongsTo` specified not to.

We probably also need a test case for this.